Error when querying dual objective value for an unbounded model with at least one unbounded variable

I have running into an `AssertError` when querying `dual_objective_value` for an unbounded model with at least one unbounded variable.
The code blow was executed on Julia 1.11.1, JuMP 1.23.5 and HiGHS v1.12.1

```julia
julia> using JuMP, HiGHS

julia> model = Model(HiGHS.Optimizer);

julia> @variable(model, x);  # unbounded variable

julia> @objective(model, Min, x);

julia> optimize!(model)  # unbounded, as expected
Running HiGHS 1.8.1 (git hash: 4a7f24ac6): Copyright (c) 2024 HiGHS under MIT licence terms
Coefficient ranges:
  Cost   [1e+00, 1e+00]
  Bound  [0e+00, 0e+00]
Solving LP without presolve, or with basis, or unconstrained
Solving an unconstrained LP with 1 columns
Model status        : Unbounded
Objective value     :  0.0000000000e+00
Relative P-D gap    :  1.0000000000e+00
HiGHS run time      :          0.00

julia> dual_objective_value(model)
ERROR: AssertionError: isfinite(u)
Stacktrace:
  [1] _active_bound
    @ ~/.julia/packages/HiGHS/P0qbU/src/MOI_wrapper.jl:2275 [inlined]
  [2] get(model::HiGHS.Optimizer, attr::MathOptInterface.DualObjectiveValue)
    @ HiGHS ~/.julia/packages/HiGHS/P0qbU/src/MOI_wrapper.jl:2228
  [3] get
    @ ~/.julia/packages/MathOptInterface/gLl4d/src/Bridges/bridge_optimizer.jl:889 [inlined]
  [4] _get_model_attribute(model::MathOptInterface.Utilities.CachingOptimizer{…}, attr::MathOptInterface.DualObjectiveValue)
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/gLl4d/src/Utilities/cachingoptimizer.jl:847
  [5] get
    @ ~/.julia/packages/MathOptInterface/gLl4d/src/Utilities/cachingoptimizer.jl:900 [inlined]
  [6] _moi_get_result(model::MathOptInterface.Utilities.CachingOptimizer{…}, args::MathOptInterface.DualObjectiveValue)
    @ JuMP ~/.julia/packages/JuMP/i68GU/src/optimizer_interface.jl:1053
  [7] get(model::Model, attr::MathOptInterface.DualObjectiveValue)
    @ JuMP ~/.julia/packages/JuMP/i68GU/src/optimizer_interface.jl:1073
  [8] dual_objective_value(model::Model; result::Int64)
    @ JuMP ~/.julia/packages/JuMP/i68GU/src/objective.jl:167
  [9] dual_objective_value(model::Model)
    @ JuMP ~/.julia/packages/JuMP/i68GU/src/objective.jl:163
 [10] top-level scope
    @ REPL[6]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

The stacktrace points to this line: https://github.com/jump-dev/HiGHS.jl/blob/6914b62a8c60bf6b4614df877fd661d2433d0dc8/src/MOI_wrapper.jl#L2275
which I believe is only triggered when hitting an unbounded variable

--- a/src/MOI_wrapper.jl
+++ b/src/MOI_wrapper.jl
@@ -2271,9 +2271,10 @@ function _active_bound(l, x, u, d)
         end
     elseif isfinite(l)
         return l
-    else
-        @assert isfinite(u)
+    elseif isfinite(u)
         return u
+    else
+        return 0.0
     end
 end
 

--- a/test/MOI_wrapper.jl
+++ b/test/MOI_wrapper.jl
@@ -941,11 +941,13 @@ function test_active_bound()
         (0.0, 1.0, 1.0, -2.0) => 1.0,
         (0.0, 0.6, 1.0, -2.0) => 1.0,
         (0.0, 0.6, 1.0, 2.0) => 1.0,  # incorrect d but doesn't matter
+        (-Inf, 0.0, Inf, 0.0) => 0.0,
         # It's a ray. Choose based on sign
         (0.0, NaN, 1.0, 2.0) => 0.0,
         (0.0, NaN, 1.0, 1e-10) => 0.0,
         (0.0, NaN, 1.0, -2.0) => 1.0,
         (0.0, NaN, 1.0, -1e-10) => 1.0,
+        (-Inf, NaN, Inf, 0.0) => 0.0,
         # It's a one-sided ray
         (0.0, NaN, Inf, 2.0) => 0.0,
         (0.0, NaN, Inf, -1e-10) => 0.0,

--- a/test/MOI_wrapper.jl
+++ b/test/MOI_wrapper.jl
@@ -974,6 +974,18 @@ function test_add_constrained_variable_tuple()
     return
 end
 
+function test_dual_objective_value_infeasible()
+    model = HiGHS.Optimizer()
+    x = MOI.add_variable(model)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    f = 1.0 * x
+    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+    MOI.optimize!(model)
+    @test MOI.get(model, MOI.DualStatus()) == MOI.INFEASIBLE_POINT
+    @test MOI.get(model, MOI.DualObjectiveValue()) == 0.0
+    return
+end
+
 end  # module
 
 TestMOIHighs.runtests()

